Orange/preprocess/randomize.py

```diff
@@ -0,0 +1,37 @@
+import numpy as np
+
+from Orange.data import Table
+from .preprocess import Randomize
+
+__all__ = ["Randomizer"]
+
+
+class Randomizer:
+    def __init__(self, rand_type=Randomize.RandomizeClasses):
+        self.rand_type = rand_type
+
+    def __call__(self, data):
+        domain = data.domain
+        X = data.X.copy()
+        Y = data.Y.copy()
+        metas = data.metas.copy()
+
+        if self.rand_type == Randomize.RandomizeClasses:
+            Y = self.randomize(Y)
+        elif self.rand_type == Randomize.RandomizeAttributes:
+            X = self.randomize(X)
+        elif self.rand_type == Randomize.RandomizeMetas:
+            metas = self.randomize(metas)
+        else:
+            raise TypeError('Unsuported type')
+
+        return Table(domain, X, Y, metas)
+
+    def randomize(self, table):
```

Comment:
This method does not need to return anything.
If it does not it is clearer that it has side-effects.